dask/array/creation.py

```diff
@@ -210,23 +224,28 @@ def linspace(start, stop, num=50, chunks=None, dtype=None):
 
     range_ = stop - start
 
-    space = float(range_) / (num - 1)
+    div = (num - 1) if endpoint else num
+    step = float(range_) / div
 
     if dtype is None:
         dtype = np.linspace(0, 1, 1).dtype
 
-    name = 'linspace-' + tokenize((start, stop, num, chunks, dtype))
+    name = 'linspace-' + tokenize((start, stop, num, endpoint, chunks, dtype))
 
     dsk = {}
     blockstart = start
 
     for i, bs in enumerate(chunks[0]):
-        blockstop = blockstart + ((bs - 1) * space)
-        task = (partial(np.linspace, dtype=dtype), blockstart, blockstop, bs)
-        blockstart = blockstart + (space * bs)
+        bs_space = bs - 1 if endpoint else bs
+        blockstop = blockstart + (bs_space * step)
+        task = (partial(np.linspace, endpoint=endpoint, dtype=dtype), blockstart, blockstop, bs)
+        blockstart = blockstart + (step * bs)
```

Comment:
Is this needed for something?

Reply:
It's needed to update the `start` parameter for `np.linspace` that's being applied to each block